docs/understanding-tracking-design/defining-the-data-to-collect-with-data-poducts/ui/index.md

```diff
@@ -56,4 +56,50 @@ Data Products created prior to the release of [Source Applications](../../organi
\ No newline at end of file
+![Updating existing Event Specifications](images/edit-existing-event-specification.png)
+
+## Upgrading Event Specification Instructions
+
+When working with Event Specifications of a Data Product, it’s essential to account for the evolution of underlying [Data Structures](../../managing-your-data-structures/index.md). Data Structures define reusable JSON schemas, which can be referenced by different Event Specifications (events and entities). Each Event Specification may contain instructions, which relay on a specific version of a Data Structure, adding another layer to specialize or constraint Event Specifications in a more granular way.
+
+### Versioning of Data Structures
+
+As data and your events evolve, Data Structures may be updated to new versions, which can either be compatible or incompatible with previous versions of the Data Structure. These changes can lead to potential conflicts with the instructions in the Event Specification (events and entities) that reference an older version of the Data Structure.
+
+### Semi-Automatic Upgrade of Event Specifications via the UI
+
+To streamline the process of upgrading an Event Specification to the latest version of a Data Structure, we’ve implemented a mechanism that allows you to update event specification instructions. Here’s how it works:
+
+When a new version of a Data Structure becomes available, the system will indicate that the event or entities referenced by the data structure has a new version available, showing an **'Upgrade'** button in the UI. 
+
+![Upgrade Event Specification warning](images/upgrade-event-specification-warning.png)
+
+If clicked, it will navigate to a new page, informing the user of the new version they are upgrading to, along with a **'View Changes'**.
+
+![Upgrade Event Specification page](images/upgrade-event-specification-page.png)
+
+When clicked it will show the differences between the current version of the Data Structure and the one the user intends to upgrade to. 
+
+![Upgrade Event Specification diff](images/upgrade-event-specification-diff.png)
+
+At the bottom, a button will allow users to confirm the upgrade. One of two things can happen when the upgrade is confirmed:
+
+#### 1. Successful automatic upgrade
+
+- If the Event Specification instructions are compatible with the new Data Structure version, the system will automatically upgrade the Event Specification to the latest version of the Data Structure.
+- All instructions will be updated seamlessly without further user intervention.
+
+![Automatic upgrade Event Specification](images/success_upgrade.png)
+
+#### 2. Conflict detection and resolution
+
+If the new version of the Data Structure introduces incompatibilities with the existing Event Specification instructions, the system will flag the conflicting properties.
+
+- The UI will prompt the user to resolve the conflict before the Event Specification can be upgraded.
```

Comment:
"the conflict before the Event Specification can be upgraded": could be "**these** conflicts before the Event Specification can be upgraded" for smoother phrasing.